### PR TITLE
Refactored Dsc to stand-alone file, added tests and fixes

### PR DESCRIPTION
**Pull request by joachimmetz**
* Refactored Dsc to stand-alone file and added tests
* Changes to address issue in GetUuidEntryFromUuid #15
* Addressed inconsistent return values of ReadFmtStringAndEntriesFromVirtualOffset #17

### CODE REVIEW

**Comment by joachimmetz** [comment]
@ydkhatri please merge #12 and #13 first and let me know when you did so I can fix merge conflicts.

**Comment by joachimmetz** [comment]
These files appear to be +50M I'll see if I can craft a smaller one for testing.

**Comment by ydkhatri** [comment]
> @ydkhatri please merge #12 and #13 first and let me know when you did so I can fix merge conflicts.

All done.

**Comment by joachimmetz** [comment]
@ydkhatri merged changed, this PR contains fixes for #15 and #17.

File: UnifiedLog/Lib.py [hunk 1]
```diff
@@ -1009,8 +1012,9 @@ def ProcessDataChunk(self, buffer, catalog, meta_chunk_index, debug_file_pos, lo
                                     logger.debug("fmt_str_v_offset highest bit set @ 0x{:X} ct={}".format(log_file_pos, ct))
                                 else:
                                     format_str, cache_a, cache_b = dsc_cache.ReadFmtStringAndEntriesFromVirtualOffset(fmt_str_v_offset)
-                            except:
+                            except (KeyError, IOError):
                                 logger.error('Failed to get DSC msg string @ 0x{:X} ct={}'.format(log_file_pos, ct))
+
```
Comment (ydkhatri) [inline UnifiedLog/Lib.py]: Here insert in the except block
 `format_str = '<compose failure [UUID]>'`

Comment (joachimmetz) [reply UnifiedLog/Lib.py]: ack I'll do the same at line 1000

**Comment by joachimmetz** [comment]
@ydkhatri PR updated, I've made an additional change based on your input.

File: UnifiedLog/Lib.py (hunk 1 above)
Comment (ydkhatri) [reply UnifiedLog/Lib.py]: I'm not sure we should do that for the signpost_name. I do not know the default behavior on that from the log utility.

**Pull request merged**